Hunted Mobile/Hunted Mobile/Hunted Mobile/ViewModel/MainPageViewModel.cs

```diff
@@ -81,9 +83,11 @@ public class MainPageViewModel : BaseViewModel {
             SubmitButtonIsEnable = false;
             await GetAll();
 
+            Game toBeJoined = await _gameRepository.GetGame(InviteKeyModel.GameId);
+
             // Navigate when InviteKey is valid
-            if(IsValid = ValidationHelper.IsFormValid(InviteKeyModel, _page) && !IsOverlayVisible) { 
-                await NavigateToEnterUsernamePage();
+            if((IsValid = ValidationHelper.IsFormValid(InviteKeyModel, _page) && !IsOverlayVisible) || toBeJoined.Status == GameStatus.OnGoing || toBeJoined.Status == GameStatus.Paused || toBeJoined.Status == GameStatus.Finished) { 
```

Comment:
Misschien kun je die lange sliert van game status checks in een nieuw if statement doen

Reply:
Ik heb de if statement eruit gejurgolled